hera_mc/cm_redis_corr.py

```diff
@@ -0,0 +1,160 @@
+import redis
+import time
+import six
+from . import cm_sysutils
+from .correlator import DEFAULT_REDIS_ADDRESS
+
+
+def snap_part_to_host_input(part, rsession=None):
+    """
+    Given a part string, eg. 'e2>SNPC000008' it will return the putative hostname
+    and adc number.  If a redis session is supplied and key available it returns the
+    hostname, otherwise it returns the SNAP name.  If None, returns the SNAP name.
+
+    Parameters
+    ----------
+    part : str
+        port>snap part string as returned by cminfo
+    rsession : None or redis session
+        If redis session and key available it returns the hostname, otherwise SNAP name
+        If None, returns the SNAP name
+
+    Returns
+    -------
+    hostname : str
+        hostname as parsed from input 'part' and redis
+    adc_num : str
+        port ADC number, as parsed from input 'part'
+    """
+    adc, name = part.split('>')
+    adc_num = int(adc[1:]) // 2  # divide by 2 because ADC is in demux 2
+    if rsession is None:
+        hostname = name
+    else:
+        _x = rsession.hget('corr:map', 'snap_host')
+        if _x is None:
+            hostname = name
+        else:  # pragma:  no cover
+            hostname = json.loads(_x)[name]
+    return hostname, adc_num
+
+
+def cminfo_redis_snap(cminfo, rsession=None):
+    """
+    Use hera_mc's get_cminfo_correlator method to build a dictionary
+    of correlator mappings
+
+    Parameters
+    ----------
+    cminfo : dict
+        Dictionary as returned from get_cminfo_correlator()
+    rsession : None or redis session
+        Pass through of redis session
+
+    Returns
+    -------
+    snap_to_ant : dict
+        Dictionary mapping the snaps to the antennas
+    ant_to_snap : dict
+        Dictionary mapping each antenna to its snap
+    """
+    snap_to_ant = {}
+    ant_to_snap = {}
+    for antn, ant in enumerate(cminfo['antenna_numbers']):
+        name = cminfo['antenna_names'][antn]
+        for pol in cminfo['correlator_inputs'][antn]:
+            if pol.startswith('e'):
+                e_pol = pol
+            if pol.startswith('n'):
+                n_pol = pol
+        ant_to_snap[ant] = {}
+        if e_pol != 'None':
+            snapi_e, channel_e = snap_part_to_host_input(cminfo['correlator_inputs'][antn][0],
+                                                         rsession=rsession)
+            ant_to_snap[ant]['e'] = {'host': snapi_e, 'channel': channel_e}
+            if snapi_e not in snap_to_ant.keys():
+                snap_to_ant[snapi_e] = [None] * 6
+            snap_to_ant[snapi_e][channel_e] = name + 'E'
+        if n_pol != 'None':
+            snapi_n, channel_n = snap_part_to_host_input(cminfo['correlator_inputs'][antn][1],
+                                                         rsession=rsession)
+            ant_to_snap[ant]['n'] = {'host': snapi_n, 'channel': channel_n}
+            if snapi_n not in snap_to_ant.keys():
+                snap_to_ant[snapi_n] = [None] * 6
```

Comment:
This line isn't covered by tests currently.

Reply:
There were a couple of goofy things there from hera_corr_f, which I've hopefully made better.  I don't understand how that line couldn't have been covered even with the previous version.